### PR TITLE
Flask implementation of the project

### PR DESCRIPTION
**Pull request by ShahriyarR**
Flask implementation of the project.

### CODE REVIEW

File: src/onetime/entrypoints/web/Flask/.DS_Store
Comment (ShahriyarR) [inline src/onetime/entrypoints/web/Flask/.DS_Store:1]: Ignore this one as it is something related to the MAC and should not be in the repository

File: src/onetime/entrypoints/web/Flask/__init__.py
Comment (ShahriyarR) [inline src/onetime/entrypoints/web/Flask/__init__.py:1]: use only full imports

File: src/onetime/entrypoints/web/Flask/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+from .app import app
```
Comment (ShahriyarR) [inline src/onetime/entrypoints/web/Flask/__init__.py:1]: Use full imports, not relative imports.

File: src/onetime/entrypoints/web/Flask/app.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
+from flask import Flask
+from flask_cors import CORS
+from werkzeug.middleware.proxy_fix import ProxyFix
+
+
+app = Flask(__name__)
+app.config['SECRET_KEY'] = 's3cr3t_k3y'
```
Comment (ShahriyarR) [inline src/onetime/entrypoints/web/Flask/app.py:7]: secret key must be a secured random value. please see the Django project how to use encryption for the secret key generation

File: src/onetime/entrypoints/web/Flask/views.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+
+@app.route('/', methods=['GET', 'POST'])
+@inject
+def index():
+    form = SecretCreateForm()
+    if request.method == "POST":
+        form = SecretCreateForm(request.form)
+        if not form.validate_on_submit():
+            container = Container()
```
Comment (ShahriyarR) [inline src/onetime/entrypoints/web/Flask/views.py:24]: The web view must not be responsible for the DI container setup.
The dependencies must be already available beforehand for the view.
Please see Django application for the proper DI usage.

File: src/onetime/entrypoints/web/Flask/views.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+            secret_url = url_for("secret", uuid=uuid, _external=True)
+            return render_template("onetimesecrets/index.html", form=form, secret_url=secret_url)
+    
+    return render_template("onetimesecrets/index.html", form=form)
+
+
+@app.route("/secret/<string:uuid>", methods=["GET", "POST"])
+@inject
+def secret(uuid):
+    container = Container()
```
Comment (ShahriyarR) [inline src/onetime/entrypoints/web/Flask/views.py:37]: Same here, the DI Container initialization is not a responsibility of the web view.

File: src/onetime/entrypoints/web/Flask/app.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+@app.after_request
+def add_security_headers(response):
+    response.headers['Strict-Transport-Security'] = 'max-age=31536000'
+    response.headers['X-Frame-Options'] = 'SAMEORIGIN'
+    # response.headers['Content-Security-Policy'] = "default-src 'self'"
+    response.headers['X-Content-Type-Options'] = 'nosniff'
+    return response
+
+# Güvenlik ayarları
```
Comment (ShahriyarR) [inline src/onetime/entrypoints/web/Flask/app.py:45]: turkish comment?

File: src/onetime/entrypoints/web/Flask/app.py
Comment (ShahriyarR) [inline src/onetime/entrypoints/web/Flask/app.py:1]: Separate the app and settings. Use application factories:
https://flask.palletsprojects.com/en/2.3.x/patterns/appfactories/

File: src/onetime/entrypoints/web/Flask/views.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
+
+from flask import request, url_for, render_template, session
+from .app import app
+from .form import SecretCreateForm
+from dependency_injector.wiring import Provide, inject
+from onetime.configurator.containers import Container
+from onetime.use_cases.manager import SecretAndUrlManager
+
+#exceptions
```
Comment (ShahriyarR) [inline src/onetime/entrypoints/web/Flask/views.py:9]: redundant comment

File: src/onetime/entrypoints/web/Flask/views.py [hunk 7]
```diff
@@ -0,0 +1,47 @@
+
+from flask import request, url_for, render_template, session
+from .app import app
+from .form import SecretCreateForm
```
Comment (ShahriyarR) [inline src/onetime/entrypoints/web/Flask/views.py:4]: full imports, please.
please use pre-commit hooks, in the hooks, there is a functionality that makes all import to be absolute.

**Review by ShahriyarR — COMMENTED** [review]
Please update accordingly.

**Comment by ShahriyarR** [comment]
Also please check the secure and lint fails.

File: src/onetime/entrypoints/web/Flask/views.py (hunk 7 above)
Comment (vahidoqtayoglu) [reply src/onetime/entrypoints/web/Flask/views.py]: solved

File: src/onetime/entrypoints/web/Flask/views.py (hunk 6 above)
Comment (vahidoqtayoglu) [reply src/onetime/entrypoints/web/Flask/views.py]: solved

File: src/onetime/entrypoints/web/Flask/app.py (hunk 5 above)
Comment (vahidoqtayoglu) [reply src/onetime/entrypoints/web/Flask/app.py]: solved

File: src/onetime/entrypoints/web/Flask/views.py (hunk 4 above)
Comment (vahidoqtayoglu) [reply src/onetime/entrypoints/web/Flask/views.py]: solved.

A new container was written for flask.

in another, get_secret and other methods were not visible.

File: src/onetime/entrypoints/web/Flask/views.py (hunk 3 above)
Comment (vahidoqtayoglu) [reply src/onetime/entrypoints/web/Flask/views.py]: solved.

A new container was written for flask.

in another, get_secret and other methods were not visible.

File: src/onetime/entrypoints/web/Flask/app.py [hunk 8]
```diff
@@ -0,0 +1,119 @@
+from flask import Flask
+from flask_cors import CORS
+from werkzeug.middleware.proxy_fix import ProxyFix
+
+
+app = Flask(__name__)
+app.config['SECRET_KEY'] = 's3cr3t_k3y'
+
+from .views import *
```
[the remark this answers is not in the document]
Comment (vahidoqtayoglu) [reply src/onetime/entrypoints/web/Flask/app.py]: solved

File: src/onetime/entrypoints/web/Flask/app.py (hunk 2 above)
Comment (vahidoqtayoglu) [reply src/onetime/entrypoints/web/Flask/app.py]: solved

File: src/onetime/entrypoints/web/Flask/__init__.py (hunk 1 above)
Comment (vahidoqtayoglu) [reply src/onetime/entrypoints/web/Flask/__init__.py]: solved

File: src/onetime/entrypoints/web/Flask/.DS_Store
Comment (vahidoqtayoglu) [reply src/onetime/entrypoints/web/Flask/.DS_Store:1]: solved

File: src/onetime/entrypoints/web/web_flask/containers.py [hunk 9]
```diff
@@ -0,0 +1,13 @@
+from dependency_injector import containers, providers
+from onetime.services.manager import SecretManager
+from onetime.use_cases.manager import SecretAndUrlManager
+
+
+class Container(containers.DeclarativeContainer):
```
Comment (ShahriyarR) [inline src/onetime/entrypoints/web/web_flask/containers.py:6]: No need for a separate container.
Please use what we have in `configurator/contianers`

File: src/onetime/entrypoints/web/web_flask/app.py [hunk 10]
```diff
@@ -0,0 +1,105 @@
+from flask import Flask
+from flask_cors import CORS
+from werkzeug.middleware.proxy_fix import ProxyFix
+from onetime.entrypoints.web.web_flask.containers import Container
+from cryptography.fernet import Fernet
+from uuid import uuid4
+
+
+app = Flask(__name__)
```
Comment (ShahriyarR) [inline src/onetime/entrypoints/web/web_flask/app.py:9]: Please use the Flask application factory.
A similar approach as shown here:
https://github.com/ShahriyarR/hexagonal-flask-blog-tutorial/blob/main/src/blog/adapters/entrypoints/app/application.py#L9

**Review by ShahriyarR — COMMENTED** [review]
Please change as commented.
Thanks.

**Comment by ShahriyarR** [comment]
Can be considered as ready to merge.
@vahidoqtayoglu please prepare the test suite for the flask in the next PR.

**Pull request merged**